Sources/ImageFetcher/Networking.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  Networking.swift
+//  Mobelux
+//
+//  MIT License
+//
+//  Copyright (c) 2022 Mobelux LLC
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+//
+
+import Foundation
+
+/// A simple wrapper for a closure performing an async network request.
+public struct Networking {
```

Comment:
It feels like there could be some benefit to `Networking` to be a protocol, mostly around improving testability of this package. What do you think?

Reply:
I find myself slowly becoming more open towards PointFree's skepticism about protocols for dependencies. I was thinking you could do something along the lines of the following for testing:

```swift
enum Mock {
    static let data = Data()
    static let response = URLResponse()
}

extension Networking {
    static var mock = Networking { _ in
        (Mock.data, Mock.response)
    }
}

// or

var responses: [(Data, URLResponse)] = [ ... ]
let networking = Networking { _ in
    responses.popLast()!
}
```

Reply:
That makes sense. I missed this context.